### PR TITLE
0040628: Course Links require Read Access to Target Course

### PR DESCRIPTION
**Pull request by playamongthestars**
This PR is for ticket: https://mantis.ilias.de/view.php?id=40628

### CODE REVIEW

File: Modules/CategoryReference/classes/class.ilObjCategoryReferenceGUI.php [hunk 1]
```diff
@@ -1,7 +1,5 @@
 <?php
 
-declare(strict_types=1);
```
Comment (mjansenDatabay) [inline Modules/CategoryReference/classes/class.ilObjCategoryReferenceGUI.php:3]: @nmatuschek We should not remove this, but only move this below the license header.

File: Services/ContainerReference/classes/class.ilContainerReferenceGUI.php [hunk 2]
```diff
@@ -1,7 +1,5 @@
 <?php
 
-declare(strict_types=1);
```
Comment (mjansenDatabay) [inline Services/ContainerReference/classes/class.ilContainerReferenceGUI.php:3]: @nmatuschek We should not remove this, but only move this below the license header.

File: Services/ContainerReference/classes/ilContainerReferenceRepositorySelectorExplorerGUI.php [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+/**
+ * Explorer for selecting repository items.
+ *
+ * The implementation starts as a replacement for the often (ab)used ilSearchRootSelector class.
+ * Clicking items triggers a "selection" command.
+ * However ajax/checkbox/radio and use in an inputgui class should be implemented in the future, too.
+ *
+ * @author Alexander Killing <killing@leifos.de>
+ */
+class ilContainerReferenceRepositorySelectorExplorerGUI extends ilTreeExplorerGUI
```
Comment (mjansenDatabay) [inline Services/ContainerReference/classes/ilContainerReferenceRepositorySelectorExplorerGUI.php]: Please add a `class.` to the filename.

Comment (mjansenDatabay) [inline Services/ContainerReference/classes/ilContainerReferenceRepositorySelectorExplorerGUI.php]: Furthermore it should be sufficient to use `extends ilRepositorySelectorExplorerGUI` and simply **overwrite** `isNodeClickable`

File: Services/ContainerReference/classes/ilContainerReferenceRepositorySelectorExplorerGUI.php [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+
+/**
+ * Explorer for selecting repository items.
+ *
+ * The implementation starts as a replacement for the often (ab)used ilSearchRootSelector class.
+ * Clicking items triggers a "selection" command.
+ * However ajax/checkbox/radio and use in an inputgui class should be implemented in the future, too.
+ *
+ * @author Alexander Killing <killing@leifos.de>
+ */
```
Comment (mjansenDatabay) [inline Services/ContainerReference/classes/ilContainerReferenceRepositorySelectorExplorerGUI.php]: Please remove this comment.

File: Services/ContainerReference/Form/class.ilContainerReferenceRepositorySelectorInputGUI.php [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+use ILIAS\ContainerReference\ilContainerReferenceRepositorySelectorExplorerGUI;
+
+/**
+ * Select repository nodes
+ *
+ * @author Alexander Killing <killing@leifos.de>
```
Comment (mjansenDatabay) [inline Services/ContainerReference/Form/class.ilContainerReferenceRepositorySelectorInputGUI.php]:
```suggestion
```

**Comment by fsesterh** [comment]
Hi @smeyer-ilias , any feedback on this topic?

**Comment by alex40724** [comment]
Hi @nmatuschek 

Thanks for providing a PR for this issue.

Please answer the following questions:
- [x] Why having a whole new class to tackle this, couldn't we just add a configure option to ilRepositorySelector2InputGUI which permission to check (read/visible)? I suspect the resulting code change would be much smaller.

Please implement the following changes:
- [x] Please add a configure option for ilRepositorySelector2InputGUI unless there is a very good reason not to do so. We should have less code and make the option available for other contexts as well.

kindly,
@alex40724

**Review by smeyer-ilias — APPROVED** [review]
Thanks @nmatuschek  for fixing this

**Pull request merged**